src/open_samus_returns_rando/specific_patches/heat_room_patches.py

```diff
@@ -1,239 +1,101 @@
-    new_logic_shape = _get_new_logic_shape(editor, 4)
-    new_logic_shape["data"]["total_boundings"] =  total_boundings
-    new_logic_shape["data"]["polys"][0]["boundings"] =  total_boundings
-    points = new_logic_shape["data"]["polys"][0]["points"]
-
-    points[0]["x"] = 5500.0
-    points[0]["y"] = 5900.0
-    points[1]["x"] = 17700.0
-    points[1]["y"] = 5900.0
-    points[2]["x"] = 17700.0
-    points[2]["y"] = 10700.0
-    points[3]["x"] = 5500.0
-    points[3]["y"] = 10700.0
-
-    scenario_5.raw["objects_c"]["LS_Heat_002"] = new_logic_shape
-
-    heat_actor = _get_heat_trigger(editor)
-    new_actor = editor.copy_actor(name_of_scenario, (5600.0, 16000.0, 0.0), heat_actor, "TG_Heat_033", 2)
-    new_actor["components"][0]["arguments"][16]["value"] = "LS_Heat_002"
-
-    scenario_5.add_actor_to_entity_groups("collision_camera_009", "TG_Heat_033", True)
-    scenario_5.add_actor_to_entity_groups("collision_camera_015", "TG_Heat_033", True)
-    scenario_5.add_actor_to_entity_groups("collision_camera_017", "TG_Heat_033", True)
-    scenario_5.add_actor_to_entity_groups("017_PostGamma_002", "TG_Heat_033", True)
-    scenario_5.add_actor_to_entity_groups("PostGamma_002", "TG_Heat_033", True)
-
-def _patch_area_6_b(editor: PatcherEditor):
-    name_of_scenario = "s065_area6b"
-
-    scenario_6b = editor.get_scenario(name_of_scenario)
-
-    total_boundings = ListContainer([7600.0, 5800.0, 14200.0, 7900.0])
-    new_logic_shape = _get_new_logic_shape(editor, 4)
-    new_logic_shape["data"]["total_boundings"] =  total_boundings
-    new_logic_shape["data"]["polys"][0]["boundings"] =  total_boundings
-    points = new_logic_shape["data"]["polys"][0]["points"]
-
-    points[0]["x"] = 7600.0
-    points[0]["y"] = 5800.0
-    points[1]["x"] = 14200.0
-    points[1]["y"] = 5800.0
-    points[2]["x"] = 14200.0
-    points[2]["y"] = 7900.0
-    points[3]["x"] = 7600.0
-    points[3]["y"] = 7900.0
-
-    scenario_6b.raw["objects_c"]["LS_Heat_001"] = new_logic_shape
-
-    heat_actor = _get_heat_trigger(editor)
-    editor.copy_actor(name_of_scenario, (6000.0, 9000.0, 0.0), heat_actor, "TG_Heat_003", 2)
-
-    scenario_6b.add_actor_to_entity_groups("collision_camera_014", "TG_Heat_003", True)
-    scenario_6b.add_actor_to_entity_groups("collision_camera_017", "TG_Heat_003", True)
-    scenario_6b.add_actor_to_entity_groups("collision_camera_018", "TG_Heat_003", True)
-    scenario_6b.add_actor_to_entity_groups("PostGamma_002", "TG_Heat_003", True)
+class NewHeatActor(typing.NamedTuple):
+    scenario: str
+    position: list[float]
+    total_boundings: list[float]
+    points: list[tuple[float, float]]
+    entity_groups: list[str]
+    trigger_name: str = "TG_Heat_Rando_001"
+    logic_shape_name: str = "LS_Heat_Rando_001"
+
+new_heat_actors = [
+    NewHeatActor(
+        "s033_area3b", [1000.0, 5000.0, 0.0], [-500.0, 3800.0, 3900.0, 6700.0],
+        ([-500.0, 3800.0], [3900.0, 3800.0], [3900.0, 6700.0], [-500.0, 6700.0]),
```

Comment:
Dynamically typed languages are fun 😀
The type in line 19 is correct but you forgot to change the values e.g. here in 27.
The `(` +`)` define a tuple and the `[` + `]` define a list. To match the type they should be used the exact opposite way.
To get `points: list[tuple[float, float]]` it should be `[(-500.0, 3800.0), (3900.0, 3800.0), (3900.0, 6700.0), (-500.0, 6700.0)],`

But it already works because you can access a tuple and a list the same way / with the same syntax.
Hence: Dynamically typed languages are fun